src/compile/scale/domain.ts

```diff
@@ -267,6 +267,15 @@ export function domainSort(model: UnitModel, channel: ScaleChannel, scaleType: S
 
   const sort = model.sort(channel);
 
+  // if the sort is specified with array, use the derived sort index field
+  if (isSortArray(sort)) {
+    return {
+      op: 'min',
+      field: `${channel}_${model.vgField(channel)}_sort_index`,
```

Comment:
Here you use `model.vgField(channel)` but in the calculate.ts file you use `${channel}_{fieldDef.field}_sort_index`.  

For the most cases, this should be fine, but I wonder if there are cases where we apply aggregation function for categorical value -- in such case `model.vgField(channel)` would return different result from `fieldDef.field`.

I think we should have a helper method in `calculate.ts` for generating this field name (from a `fieldDef`?) and make both calculate.ts and the code here call the same method for safety in the future?

Reply:
Should we stick to `vgField` or just `fieldDef.field` as a field name?

Reply:
`vgField(fieldDef)` ?